interactive/dashboard/flow_diagram.py

```diff
@@ -119,153 +94,1008 @@ def name_included_reactions(included_reactions, reactions_json):
+
+            li = species_yields
+            logged = []
+            for i in li:
+                # fail safe for null->null value
+                if (str(i) != 'null->null'
+                        and float(0.0) != float(species_yields[i])):
+                    rawVal.update({i: species_yields[i]})
+                    try:
+                        logged.append((i, math.log(species_yields[i])))
+                        if species_yields[i] < minVal:
+                            minVal = species_yields[i]
+                        if species_yields[i] > maxVal:
+                            maxVal = species_yields[i]
+                    except ValueError:
+                        numOfInvalids = numOfInvalids+1
+                        logged.append(
+                            (i, abs(math.log(abs(species_yields[i])))))
+                        if abs(species_yields[i]) < minVal:
+                            minVal = abs(species_yields[i])
+                        if abs(species_yields[i]) > maxVal:
+                            maxVal = abs(species_yields[i])
+
+            vals = [i[1] for i in logged]
+            min_val = abs(min(vals, default="EMPTY"))
+            range = max(vals) - min(vals, default="EMPTY")
+            if max(vals) == min(vals, default="EMPTY"):
+                range = 1
+            scaled = [(x[0], (float(((x[1] + min_val)/range))
+                       * float(maxWidth)) + 1) for x in logged]
+            print("|_ found", numOfInvalids,
+                  "invalid values, reversed the lines for them")
+            print("[6/7] got logarithmically scaled weights")
+            return list(scaled), minVal, maxVal, rawVal
+        else:
+            print("[7/7] no species selected, returning empty list")
+            return list([]), -1, 999999999, {}
+    else:
+        print("[5/7] linearly scaling weights...")
+        li = species_yields
+        vals = [species_yields[i] for i in li]
+        min_val = abs(min(vals, default="EMPTY"))
+
+        minVal = min_val
+        maxVal = abs(max(vals, default="EMPTY"))
+
+        range = max(vals) - min(vals, default="EMPTY")
+        scaled = [(x, (((species_yields[x] + min_val)/range)
+                   * int(maxWidth)) + 1) for x in li]
+        print("[6/7] got linearly scaled weights")
+        return list(scaled), minVal, maxVal, {}
+
+
+# 1) load reactions from output file, and create a list of reactions
+# 2) load species and reaction rates
+# 3) scale arrows and handle colors for arrows outside of range
+# 4) create nodes and edges for graph
+def new_find_reactions_and_species(list_of_species, reactions_data,
+                                   blockedSpecies, df, start,
+                                   end, max_width, scale_type,
+                                   csv_results_path_default=csv_results_path_default):
+
+    global minAndMaxOfSelectedTimeFrame
+    global userSelectedMinMax
+    global previous_vals
```

Comment:
these are not used or modified here. Remove? Also the messages with the numbers below can be info, the rest can be debug